Limit the number of parallel link jobs during LLVM build for mingw.

michaelwoerister opened the pull request:
This PR is an attempt to unblock https://github.com/rust-lang/rust/pull/108355, which keeps failing while trying to link various LLVM artifacts on mingw runners. It looks like doing too many linking jobs might put too much load on the system? (Although I don't understand why the jobs are only failing for #108355 while they seem to pass for others)


r? infra-ci

jyn514 commented on src/ci/github-actions/ci.yml:
```diff
@@ -633,7 +636,9 @@ jobs:
-              RUST_CONFIGURE_ARGS: --build=i686-pc-windows-gnu
+              RUST_CONFIGURE_ARGS: >-
+                --build=i686-pc-windows-gnu
+                --set llvm.link-jobs=$MINGW_LLVM_LINK_JOBS
```
please set this in one place around https://github.com/rust-lang/rust/blob/5e90940a4b742e7f1e86f21c26b56e99a8733458/src/ci/run.sh#L59-L60 rather than duplicating it across all jobs.

michaelwoerister replied:
Thanks for the pointer, I'll look into it 🙂

michaelwoerister committed:
Limited the number of parallel link jobs during LLVM build for mingw.

This is an attempt to fix the spurious build error tracked by
https://github.com/rust-lang/rust/issues/108227.

Mark-Simulacrum commented:
cc https://github.com/rust-lang/rust/issues/108227 which is the tracking issue for this spurious failure

I don't mind trying this, but I think we should link the tracking issue for the spurious failure from the code for the future visitor who is confused. r=me with that added.

Mark-Simulacrum commented:
@rustbot author

michaelwoerister commented:
Thanks, @Mark-Simulacrum!

@bors r=Mark-Simulacrum

bors commented:
:pushpin: Commit 40fc42371f04316e882cdfe8bf39b9dd194343ac has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 40fc42371f04316e882cdfe8bf39b9dd194343ac -->
<!-- homu: {"type":"Approved","sha":"40fc42371f04316e882cdfe8bf39b9dd194343ac","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

michaelwoerister commented:
@bors p=1 since this might potentially resolve https://github.com/rust-lang/rust/issues/108227, which is slowing down the whole build queue.

bors commented:
:hourglass: Testing commit 40fc42371f04316e882cdfe8bf39b9dd194343ac with merge 6502613a8199cab4cb68b08c7ddce6f210d8d69c...
<!-- homu: {"type":"BuildStarted","head_sha":"40fc42371f04316e882cdfe8bf39b9dd194343ac","merge_sha":"6502613a8199cab4cb68b08c7ddce6f210d8d69c"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4488207538/jobs/7895311769)
Approved by: Mark-Simulacrum
Pushing 6502613a8199cab4cb68b08c7ddce6f210d8d69c to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4488207538/jobs/7895311769"},"merge_sha":"6502613a8199cab4cb68b08c7ddce6f210d8d69c"} -->
